dvc/repo/brancher.py

```diff
@@ -41,6 +41,8 @@ def brancher(  # noqa: E302
         if all_tags:
             revs.extend(scm.list_tags())
 
+    revs = filter(None, revs)
```

Comment:
Why is this one needed in this PR?

Reply:
To prevent passing `None` arguments (e.g. `revs=["HEAD", None]`) in the following statement:

```python
for branch in self.brancher(revs=[a_ref, b_ref]):
```

That line would need to exist either on `diff` or `brancher` and I decided moving it to `brancher`, since the reader would have more context to understand what would happen if you pass `None` inside the `revs`, otherwise I found myself explaining why I was filtering `None` on `diff` and it looked out of place.